lambda/MAS/Services/StaticWebsiteService.cs

```diff
@@ -28,55 +30,87 @@ public class S3StaticWebsiteService : IStaticWebsiteService
         private readonly IAmazonS3 _amazonS3;
         private readonly ILogger<S3StaticWebsiteService> _logger;
         private readonly AWSConfig _awsConfig;
+        private readonly IAmazonCloudFront _cloudFrontService;
+        private readonly EnvironmentConfig _environmentConfig;
+        private readonly CloudFrontConfig _cloudFrontConfig;
 
-        public S3StaticWebsiteService(IAmazonS3 amazonS3, ILogger<S3StaticWebsiteService> logger, AWSConfig awsConfig)
+        public S3StaticWebsiteService(IAmazonS3 amazonS3, 
+            IAmazonCloudFront cloudFrontService, 
+            ILogger<S3StaticWebsiteService> logger,
+            AWSConfig awsConfig, 
+            EnvironmentConfig environmentConfig,
+            CloudFrontConfig cloudFrontConfig)
         {
             _amazonS3 = amazonS3;
             _logger = logger;
             _awsConfig = awsConfig;
+            _cloudFrontService = cloudFrontService;
+            _environmentConfig = environmentConfig;
+            _cloudFrontConfig = cloudFrontConfig;
         }
 
         #endregion
 
-        public async Task<HttpStatusCode> WriteFileAsync(string filePath, string contentBody)
+
+        public async Task<HttpStatusCode> WriteFilesAsync(params StaticContentRequest[] requests)
         {
-            PutObjectRequest request = new PutObjectRequest()
-            {
-                BucketName = _awsConfig.BucketName,
-                Key = filePath,
-                ContentBody = contentBody
-            };
+            HttpStatusCode responseCode = HttpStatusCode.OK;
+            var taskDict = new Dictionary<StaticContentRequest, Task<PutObjectResponse>>();
+            foreach(var req in requests)
+                taskDict.Add(req, WriteFileAsync(req));
 
-            try
+            foreach(var task in taskDict)
             {
-                var response = await _amazonS3.PutObjectAsync(request);
-                return response.HttpStatusCode;
-            }
-            catch (Exception e)
-            {
-                _logger.LogError(e, $"Failed to write ${filePath} to S3: {e.Message}");
-                throw e;
+                responseCode = (await task.Value).HttpStatusCode;
+                if (responseCode != HttpStatusCode.OK)
+                {
+                    _logger.LogError($"Writing {task.Key} resulted in a status code of {responseCode}");
+                    return responseCode;
+                }
             }
+
+            if (_environmentConfig.Name != "local" && requests.Count() > 0)
```

Comment:
I wonder if this is better to be a config option rather than just saying locally. As in, we could have an environment where we just have S3 with no CloudFront, or we could run E2E tests in Docker which might use MinIO to mimick S3 without CloudFront, in which case we'd need to be able to disable CloudFront invalidation